kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2693,6 +2695,79 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val ENGINE_DEPLOY_MODE: ConfigEntry[String] =
+    buildConf("kyuubi.engine.deploy.mode")
+      .doc("Configures the engine deploy mode, The value can be 'local', 'yarn'. " +
+        "In local mode, the engine operates on the same node as the KyuubiServer. " +
+        "In YARN mode, the engine runs within the Application Master (AM) container of YARN. " +
+        "Currently, only Hive engine supports YARN mode.")
```

Comment:
> Currently, only Hive engine supports YARN mode.

 Spark?

Reply:
I just thought about the terminology "YARN mode", should we call it "YARN cluster mode"? to align with Spark.

Reply:
Why not creating hive-specific configurations?

Reply:
YARN mode is a common running mode, and we plan to implement this mode on JDBC, TRINO engins in the future

Reply:
Yeah, but it turns out it's much more common if having engine-specific configurations to control the different engine deployments separately.

Reply:
Thanks for your suggestion, I agree with you.